pyirf/cut_optimization.py

```diff
@@ -17,38 +18,95 @@ def optimize_gh_cut(
     signal,
     background,
     reco_energy_bins,
-    gh_cut_values,
+    gh_cut_efficiencies,
     theta_cuts,
-    op,
+    op=operator.ge,
     background_radius=1 * u.deg,
     alpha=1.0,
     progress=True,
+    **kwargs
 ):
     """
-    Optimize the gh-score in every energy bin.
-    Theta Squared Cut  should already be applied on the input tables.
+    Optimize the gh-score cut in every energy bin of reconstructed energy
+    for best sensitivity.
+
+    This procedure is EventDisplay-like, since it only applies a
```

Comment:
Now, this I think puts us in a weird position.

Either we rename this function by adding an `_EventDisplay` suffix or it should be added as a section of the EventDisplay script example.

Reply:
Why do you think that? I think only optimizing the gh cut is avalid choice.

We can add other procedures and name them accordingly. But why do you think that mentioning that this procedure of optimizing the cuts is EventDisplay like is weird?

Reply:
No, I meant that it is weird if we do _not_ specify that is specific to EventDisplay (in some way, e.g. specifying it from the function name) because as the docstring says now, it is not a general API function as we wanted them to be.

Reply:
Why isn't it?

Reply:
The function says what it does, it is nothing specific to event display. It is just similar to what event display does.

Reply:
I would then do 2 things:
- simplify the docstring (no need to mention event display if there is no reference of some kind)
- rename the function in something more coherent with the doctring like `apply_theta_cut_optimize_gh_cut` or `optimize_gh_cut_with_thetacut`

because there will be other ways to optimize the cuts (different combinations)

Reply:
You have to apply a theta cut to optimize a point source sensitivity so this is nothing special. The special thing is that this only optimizes the gh cut and takes a precomputed theta cut.

I don't see why the docstring shouldn't mention that this is what EventDisplay does. It is a fact and it is also the reason why we implemented this function in the first place.

When doing the full optimization in theta / gh / multiplicity, I am also going to mention that this is how MARS does it.